app/models/sushi/item_report.rb

```diff
@@ -45,8 +45,9 @@ def initialize(params = {}, created: Time.zone.now, account:)
 
     def validate_item_report_parameters(params:, account:)
       coerce_metric_types(params, allowed_types: ALLOWED_METRIC_TYPES)
-      validate_item_id(params) if params[:item_id]
-      validate_platform(params, account) if params[:platform]
+      validate_access_method(params) if params.key?(:access_method)
```

Comment:
Hmm…there's a coupling of knowledge between the validate methods and the conditionals here.

Namely, we are assuming knowledge of the hash key here and inside the validate methods.  What if the validate methods were to instead introduce the guard clause?

Here we would have:

```ruby
validate_item_id(params)
```

Then elsewhere

```ruby
def validate_item_id(params = {})
  return true unless params.key?(:item_id)
  ...
```

Then there's reduced duplication of knowledge regarding the key that we're using for the `item_id`

Reply:
I considered adding a guard clause too....but then wondered if it would be "better" to skip entering the method at all, vs calling it and potentially immediately exiting. but I do like your point about the duplication of knowledge.

Reply:
Another thing to consider.  If I chose to locally override the method, would I want the knowledge of the guard clause within that method or outside of the method?  My preference is within.

Also, method calls are very inexpensive compared to our cognitive load of reading the initialize method, then holding that "if" statement while we go and read the validation method (remembering the "if" statement)